inst/include/Rcpp/String.h

```diff
@@ -329,11 +347,46 @@ namespace Rcpp {
             return std::wstring( s, s + strlen(s) );
         }
 
-
         inline const char* get_cstring() const {
             return buffer_ready ? buffer.c_str() : CHAR(data) ;
         }
 
+        inline const std::string get_encoding() const {
+            switch (enc) {
+                case CE_BYTES:
+                    return "bytes";
+                case CE_LATIN1:
+                    return "latin1";
+                case CE_UTF8:
+                    return "UTF-8";
+                default:
+                    return "unknown";
+            }
+        }
+
+        inline void set_encoding( cetype_t encoding ) {
+            enc = encoding;
+
+            if (valid) {
+                data = Rf_mkCharCE(Rf_translateCharUTF8(data), encoding);
```

Comment:
Okay, agreed. I think we are good to merge (if I hear no objections by
tomorrow AM I'll go ahead and merge).

On Wed, Jul 1, 2015 at 6:19 PM, Kevin Ushey <notifications@github.com>
wrote:

> In inst/include/Rcpp/String.h
> <https://github.com/RcppCore/Rcpp/pull/310#discussion_r33695590>:
>
> > +                case CE_BYTES:
> > +                    return "bytes";
> > +                case CE_LATIN1:
> > +                    return "latin1";
> > +                case CE_UTF8:
> > +                    return "UTF-8";
> > +                default:
> > +                    return "unknown";
> > +            }
> > +        }
> > +
> > +        inline void set_encoding( cetype_t encoding ) {
> > +            enc = encoding;
> > +
> > +            if (valid) {
> > +                data = Rf_mkCharCE(Rf_translateCharUTF8(data), encoding);
>
> Because the underlying SEXP is a CHARSXP, I think its lifetime is
> essentially 'infinite' since once a string enters the string pool R won't
> clean it out. At least, that's the assumption the String class is making
> all over the place; this probably won't be safe assumption 'some day' but
> is probably permissible for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RcppCore/Rcpp/pull/310/files#r33695590>.
>